common-npm-packages/artifacts-common/EntraWifUserServiceConnectionUtils.ts

```diff
@@ -2,76 +2,96 @@ import path = require("path");
 import * as tl from 'azure-pipelines-task-lib/task';
 import { getSystemAccessToken } from "./webapi";
 import fetch from "node-fetch";
+import { retryOnException } from "./retryUtils";
 
 tl.setResourcePath(path.join(__dirname, 'module.json'), true);
 
 const ADO_RESOURCE : string = "499b84ac-1321-427f-aa17-267ca6975798/.default";
 const CLIENT_ASSERTION_TYPE : string = "urn:ietf:params:oauth:client-assertion-type:jwt-bearer";
 const GRANT_TYPE = "client_credentials";
+const PPE_HOSTS : string [] = ["vsts.me","codedev.ms","devppe.azure.com"];
 
-export async function getFederatedWorkloadIdentityCredentials(serviceConnectionName: string, tenantId?: string) : Promise<string | undefined>
-{
-    try {
-        let tenant = tenantId ?? tl.getEndpointAuthorizationParameterRequired(serviceConnectionName, "TenantId");
-        tl.debug(tl.loc('Info_UsingTenantId', tenantId));
-        const systemAccessToken = getSystemAccessToken();
-        const url = process.env["SYSTEM_OIDCREQUESTURI"]+"?api-version=7.1&serviceConnectionId="+serviceConnectionName;
-        
-        const ADOResponse: {oidcToken: string} = await (await fetch(url, 
-        {
+export async function getFederatedWorkloadIdentityCredentials(serviceConnectionName: string, tenantId?: string) : Promise<string | undefined>{
+    let tenant = tenantId ?? tl.getEndpointAuthorizationParameterRequired(serviceConnectionName, "TenantId");
```

Comment:
getEndpointAuthorizationParameterRequired() will throw if the required parameter isn't set and has now been moved outside of the try block. Maybe we should add a comment so callers are aware of this and will handle the throw?

Reply:
It prints a nice error message with it already! I am planning to leave as is.